demos-europe/demosplan-core
# refactor(refs T19759): make filter modal work again [vue 3 upgrade]

**Integration branch:** #1151
**Ticket:** https://yaits.demos-deutschland.de/T19759



Fixes the filter modal component in the assessment table:

- remove `portal-vue` as it does not seem to be needed anymore
- remove `vue-multiselect` from dependencies (it is included in demosplan-ui)
- use `defineAsyncComponents` on async components, see https://v3-migration.vuejs.org/breaking-changes/async-components.html
- adjust filter modal (and other usages of DpTab(s)) after changes to DpTabs/DpTab in demosplan-ui, see https://github.com/demos-europe/demosplan-ui/pull/650
- use `.splice()` instead of `delete` on arrays

### How to review/test

For testing purposes, I linked to the most recent commit in the corresponding demosplan-ui branch in the package.json (this needs to be removed before merging), so you can just
- install dependencies, build demosplan-ui in dplan, run `yarn dev:<project>`
- open the project and navigate to the assessment table
- open the filter modal - it should work as expected

### Linked PRs 


- [x] #2322 (needs to be merged first)
- [x] https://github.com/demos-europe/demosplan-ui/pull/650 (needs to be merged first)

### PR Checklist
